src/pages/ServiceDetails/ServiceInfo.tsx

```diff
@@ -54,6 +62,65 @@ class ServiceInfo extends React.Component<ServiceDetails, ServiceInfoState> {
     return sorted;
   }
 
+  validationChecks(): ValidationChecks {
+    let validationChecks = {
+      hasRouteRuleChecks: false,
+      hasDestinationPolicyChecks: false,
+      hasVirtualServiceChecks: false,
+      hasDestinationRuleChecks: false
+    };
+
+    const routeRules = this.props.serviceDetails.routeRules || [];
+    const destinationPolicies = this.props.serviceDetails.destinationPolicies || [];
+    const virtualServices = this.props.serviceDetails.virtualServices || [];
+    const destinationRules = this.props.serviceDetails.destinationRules || [];
+
+    for (let i = 0; i < routeRules.length; i++) {
```

Comment:
What do you think about using [`Array.some()`](https://www.tutorialspoint.com/typescript/typescript_array_some.htm) method?

Reply:
Ah, it's like `Stream.anyMatch()` in java, good to know

Reply:
Same in Ruby `collection.any?`, pretty powerful.
Also in TS, there is the `collection.all?` method: [`Array.every()`](https://www.tutorialspoint.com/typescript/typescript_array_every.htm). It returns true if all the elements in the array match the condition passed by param. Powerful too.

Reply:
I am not a big fan of getting rid of the old school 'for' loop :-).
If the same file has some code using them, normally I would maintain it, but on this case, as it was the first loop I change it.